### ISSUE TITLE
add typedoc support

### ISSUE THREAD

**Issue by long76** [ant-design/ant-design#41665]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/peaceful-violet-p9iwxy?file=%2Fpackage.json&selection=%5B%7B%22endColumn%22%3A26%2C%22endLineNumber%22%3A22%2C%22startColumn%22%3A26%2C%22startLineNumber%22%3A22%7D%5D)

### Steps to reproduce

add typedoc to your ts project with antd
add `typedoc.json`
```json
{
  "entryPoints": ["./index.tsx"],
  "excludeExternals": true,
  "out": "docs"
}
```
*Note* Don't forgot set entry points if it's different
add in `package.json` script
```json
"docs":  "typedoc",
```
execute
```shell
npm run docs
```

### What is expected?

typedoc without error

### What is actually happening?

```javascript
node_modules/antd/es/config-provider/context.d.ts:3:30 - error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("scroll-into-view-if-needed")' call instead.

3 import type { Options } from 'scroll-into-view-if-needed';
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/antd/es/config-provider/index.d.ts:3:30 - error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("scroll-into-view-if-needed")' call instead.

3 import type { Options } from 'scroll-into-view-if-needed';
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/antd/es/form/Form.d.ts:4:30 - error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("scroll-into-view-if-needed")' call instead.

4 import type { Options } from 'scroll-into-view-if-needed';
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/antd/es/form/interface.d.ts:2:47 - error TS1479: The current file is a CommonJS module whose imports will produce 'require' calls; however, the referenced file is an ECMAScript module and cannot be imported with 'require'. Consider writing a dynamic 'import("scroll-into-view-if-needed")' call instead.

2 export type { Options as ScrollOptions } from 'scroll-into-view-if-needed';
                                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

| Environment | Info |
| --- | --- |
| antd | 5.4.0 |
| React | 18.2.0 |
| System | Kubuntu 22.04 |
| Browser | Chrome 108.0.0.0 |

**Comment by long76** [comment]
in codesandbox just restart task start

**Comment by long76** [comment]
![image](https://user-images.githubusercontent.com/18124433/230292169-aeecdb92-b79c-4bb5-87f8-b7271e0ea5a8.png)

**Comment by long76** [comment]
@afc163 
1) `scroll-into-view-if-needed` was released `3.0.7` with fix `moduleResolution` in `tsconfig.json` how i know i need wait when semver find new version and it fix error?
2) what about add simple test on succesful exec `typedoc`?

**Comment by long76** [comment]
obsolete
